Oct 2020 version

yuqiww opened the pull request:
This PR includes:
(the changes are made to `Build Standard Database.R` instead of `Build Standard Database.Rmd`)

1. modified a few places so that the code and accommodate new surveys (e.g. tour purpose, weekpart, race, etc.)
2. added checks and temporary fixes to deal with edge cases (e.g. one home lat/lon is joined to two MAZ zones)
3. added new variables to represent "bike" separately from "drive" in access/egress mode
4. added AC Transit (2017 including Dumbarton)
5. added ACE (2019) - still not finished yet due to some issues in the raw data
6. updated `Dictionary for Standard Database`
7. updated `canonical_route_crosswalk`
8. added "survey_year" to variable dictionary join in case one operator has multiple surveys in different years

@shimonisrael

DavidOry committed:
move .bat to notebook

DavidOry committed:
implementation check

yuqiww committed:
Update canonical_route_crosswalk.csv

Add routes from Solano County survey

yuqiww committed:
Update Dictionary for Standard Database.csv

Add survey variable dictionary for Solano County data

yuqiww committed:
Merge branch 'master' of https://github.com/BayAreaMetro/onboard-surveys

yuqiww committed:
add ACE survey variables to Standard Dictionary

yuqiww committed:
add ACE to canonical_route_crosswalk

yuqiww committed:
revise "generic_response" to be consistent with the standard

The "generic_response" in the original meta data is not always consistent with the standard (e.g. upper vs. lower case, point of time vs. time range), resulting in failures in the survey dataframe mutation.

yuqiww committed:
add survey year to Dictionary in case one operator has multiple surveys

yuqiww committed:
add 2018 time field to dictionary

yuqiww committed:
comment out deprecated get_rail_name function

yuqiww committed:
add survey year to standard dictionary filter

yuqiww committed:
add AC Transit, ACE to test run along with Muni

yuqiww committed:
add race variable for ACE

The same code applies to Solano County survey

yuqiww committed:
add survey_year to Standard dictionary join

yuqiww committed:
add school refinement for dest_purp

because dest_purp == "grad_school" / "college" is used later

yuqiww committed:
count 'work-related' orig_purp and dest_purp also as work tour

yuqiww committed:
accommodate different field names and data types for "vehicles_other" and "wokers_other"

yuqiww committed:
add comment to the additional vehicles_other and workers_other code

yuqiww committed:
"work before OR after, not AND" in classifying tour purpose

yuqiww committed:
accommodate other types of missing agency data

yuqiww committed:
convert "access_mode" caconical responses to standard options

yuqiww committed:
convert "egress_mode" caconical responses to standard options

Similar to "access_mode", "motorcycle" is converted to "bike"

yuqiww committed:
add comments to debug_transfers

yuqiww committed:
update canonical route and operator names so that they can be correctly processed by the code

yuqiww committed:
make race categorization applicable to surveys whose dummy format is different from SF Muni

yuqiww committed:
update race-related canonical variable names to be compatible with the processing script

yuqiww committed:
remove unnecessary code

yuqiww committed:
make the Generic_Response for hispanic consistent across surveys

yuqiww committed:
fix typo in ACE time_string (12:30PM instead of 12:30 AM)

yuqiww committed:
handle missing date in non-Bart surveys

yuqiww committed:
add new variable to separate access/egress mode "bike" from "D" (drive)

Consider "bike" as its own category "B" in the previous variables "path_access", "path_egress" and "path_label"; create new variables "path_access_comb", "path_egress_comb" and "path_label_comb" which combines "B" and "D" (drive) as "D".

yuqiww committed:
add duplicates checking for taz/maz assignments

yuqiww committed:
add a temporary fix for duplicates caused by maz spacial join

yuqiww committed:
bug fix: missing %>%

yuqiww committed:
add path_access_comb and path_egress_comb to survey_decomposition output

yuqiww committed:
use `all_of()' to select all elements in a vector

yuqiww committed:
correct some garbled route names for Caltrain survey

yuqiww committed:
add back Bart and Caltrain surveys for test run

yuqiww committed:
add date info to ACE and revise the code to handle different date format

yuqiww committed:
update ACE survey response dictionary based on new info from Redhill

yuqiww committed:
preprocess Solano County 2017 survey data

yuqiww committed:
bring in Solano County 2017 survey

yuqiww committed:
use canonical routes created from Solano County 2017 data

yuqiww committed:
recode survey variable-standard variable crosswalk for Solano 2017 survey

yuqiww committed:
add date to exported files' names

yuqiww committed:
add Union City Transit 2017 survey canonical routes

yuqiww committed:
add Union City Transit 2017 survey Survey variable/response crosswalk

yuqiww committed:
add Union City Transit 2017 survey

yuqiww committed:
add Sonoma County Transit survey

yuqiww committed:
add Sonoma County 2018 survey routes crosswalk

yuqiww committed:
add Sonoma County 2018 survey variable dictionary and fix dictionary error for AC Transit 2018 survey

yuqiww committed:
make first_board_lat/lon and workplace_lat/lon variale names consistant

yuqiww committed:
make canonical names consistent

yuqiww committed:
add SMART 2018 survey into the processing code

yuqiww committed:
add SMART 2018 survey canonical routes

yuqiww committed:
add SMART 2018 survey standard variable crosswalk

yuqiww committed:
handle "refused" in year_born from SMART 2018 survey

yuqiww committed:
categorize 'work-related' tour_purp and 'work' tour_purp

yuqiww committed:
add pre-processing Notebook for SMART 2018 survey

yuqiww committed:
add pre-processing Notebook for Union City 2017 survey

yuqiww committed:
make workplace_lat/lon standard variable names consistent

yuqiww committed:
make gender / fare_medium standard responses consistent

yuqiww committed:
recode depart_hour and return_hour for Solano County Operators (4) survey to be consistent with other surveys

yuqiww committed:
correct errors in MUNI_N OWL and MUNI_T routes coding for Soltrans and Union City surveys

yuqiww committed:
add WETA 2019 survey to processing code

yuqiww committed:
convert values from "vehicles_other" field in "ACE" and "Solano County" surveys to standard values

yuqiww committed:
assign "auto_suff" value to 'missing' for survey responses with "missing" vehicle/worker number

yuqiww committed:
add WETA 2019 survey canonical route crosswalk

yuqiww committed:
add WETA 2019 survey survey variable/response crosswalk

yuqiww committed:
code "Muni___Metro" as "Muni___Metro" (light rail) to distinguish from "Muni___Muni" (local bus)

yuqiww committed:
change "college" in tour_purp to "university" to be consistent

yuqiww committed:
Jupyter Notebook to pre-process WETA survey data

yuqiww committed:
Jupyter Notebook to preprocess sonoma_county 2018 survey

yuqiww committed:
fix survey_year error for Napa Vine 2014 survey

yuqiww committed:
add WestCAT 2017 survey to the code

yuqiww committed:
add WestCAT 2017 survey transfer routes

yuqiww committed:
add WestCAT 2017 survey variables

yuqiww committed:
Jupyter Notebook to preprocess WestCAT 2017 survey raw data

yuqiww committed:
update WETA preprocessing Jupyter Notebook directory (from local drive to M)

yuqiww committed:
make LAVTA canonical route names consistent

yuqiww committed:
fix variable dictionry typo "survey_alight_lat/lon"

yuqiww committed:
make standard variable consistent: personal/medical -> other maintenance

yuqiww committed:
remove "race_dmy_hisp“ from Sonoma County Transit 2018 survey standard variable dict because it already has a "hispanic" variable

yuqiww committed:
make canonical route/operator names consistent for Mission Bay TMA shuttle

yuqiww committed:
correct standard variable dict for Solano County 2017 survey: "race_other_string" instead of "race_dmy_other"

yuqiww committed:
modify worker_numeric_cat and vehicle_numeric_cat method to capture all values in the survey

yuqiww committed:
add detailed notes on special cases in "debug_transfers" table

yuqiww committed:
remove unnecessary variable "race_other_string" from ACE, Solano County, WestCAT survey because they already have a categorical race variable

yuqiww committed:
rename the categoric race variable to "race_cat" for ACE, Solano County, WestCAT surveys

yuqiww committed:
correct typo "Unknown" in AC Transit 2018 survey variable eng_proficient

yuqiww committed:
revise race categorization method

 to accomodate surveys with categorical instead of dummy race variables in raw survey data (ACE, Solano County 2017, WestCAT)

yuqiww committed:
add detailed notes for Muni 2017 record w/ duplicated maz join

yuqiww committed:
correct errors in Solano County 2017 survey standard variable coding

correct the mapping for "first_board_lat/lon", "last_alight_lat/lon" standard variables

yuqiww committed:
remove empty rows

yuqiww committed:
add LAVTA 2018 survey caconical routes names

yuqiww committed:
add LAVTA 2018 survey standard variable dictionary

yuqiww committed:
add LAVTA 2018 survey to build_standard_database.R code

yuqiww committed:
Jupyter Notebook to preprocess LAVTA 2018 survey data

yuqiww committed:
add TriDelta 2019 survey standard variable dictionary

yuqiww committed:
add TriDelta 2019 survey canonical route names

yuqiww committed:
add TriDelta 2019 survey to build_standard_database.R

yuqiww committed:
Jupyter Notebook to preprocess TriDelta 2019 survey raw data

yuqiww committed:
make "WESTCAT" canonical name consistent

yuqiww committed:
add County Connection 2019 survey to build_standard_database.R

yuqiww committed:
add County Connection 2019 survey canonical route names

yuqiww committed:
add County Connection 2019 survey standard variable dictionary

yuqiww committed:
Jupter Notebook to preprocess County Connection 2019 survey data

yuqiww committed:
fix error in standard dictionary: at work/not at work after surveyed trip

yuqiww committed:
add missing "technology" for Sonoma_Marin Area Rail 2018 survey

yuqiww committed:
add Golden Gate Transit (2018) survey to the code

yuqiww committed:
deal with cases where the survey responses miss 'date_string' but have 'weekpart' values

yuqiww committed:
add Golden Gate Transit (2018) transfer routes to canonical routes

yuqiww committed:
add Golden Gate Transit (2018) variables to standard variable dictionary

yuqiww committed:
fix error in County Connection (2019) variable coding: 'egress_mode', 'persons'

yuqiww committed:
fix error in LAVTA (2018) variable coding: 'persons', 'workers', 'gender'

yuqiww committed:
fix error in TriDelta (2019) variable coding: 'persons', 'orig_purp', 'workers'

yuqiww committed:
fix error in Sonoma County (2018) variable coding: 'race_dmy', 'return_hour', 'work_status'

yuqiww committed:
fix error in WETA (2019) variable coding: 'return_hour'

yuqiww committed:
fix errors in Union City (2017) variable coding: 'weekpart', 'workers'

yuqiww committed:
fix error in SMART (2018) variable coding: 'race', 'student_status', 'strata'

yuqiww committed:
make 'generic_variable' name consistent for 'time_period'

yuqiww committed:
add variable_dictionary QA/QC code to pre-processing

yuqiww committed:
fix error in variable_dictionary QA/QC code in pre-processing

yuqiww committed:
Jupyter Notebook to preprocess Golden Gate Transit (2018) survey

yuqiww committed:
add Napa Vine 2019 survey to the R code

yuqiww committed:
add Napa Vine 2019 survey canonical routes

yuqiww committed:
add Napa Vine 2019 survey variable dictionary

yuqiww committed:
Jupyter Notebook to preprocess Napa Vine 2019 survey data

yuqiww committed:
make ACE 2019 generic_response consistent for "orig/dest_purp" - "personal/medical" as "other maintenance"

yuqiww committed:
survey data dictionary v1

yuqiww committed:
change column names to make them unique

yuqiww committed:
revise AC Transit (2018) "workers" and "student_status" variable standard dictionary

yuqiww committed:
Jupyter notebook (Python) to QA/QC raw survey data

yuqiww committed:
code "ACE" as "commuter rail"

yuqiww committed:
code "shuttle" technology as "local bus" in canonical route dictionary (for transfers)

yuqiww committed:
add additional variables and rename the standard variable dictionary file

yuqiww committed:
code "airport" and "hotel" as "other discretionary" for orig_purp/dest_purp

yuqiww committed:
add Smart (2018) "survey_route" and detailed access_mode/egress_mode

yuqiww committed:
preprocess new SMART (2018) survey data from consultant

shimonisrael reviewed (commented):
@yuqiww, I want to be sure I understand the update canonical_route_crosswalk.csv  is done. Do you take the 3 transfer from and 3 transfer to routes and put them into the canonical route crosswalk and then manually append the crosswalk route name?

yuqiww commented:
> @yuqiww, I want to be sure I understand the update canonical_route_crosswalk.csv is done. Do you take the 3 transfer from and 3 transfer to routes and put them into the canonical route crosswalk and then manually append the crosswalk route name?

Yes, for each survey, I collected all the different transfers (operator+route), and assigned them 'canonical_name', 'canonical_operator' and 'technology' by referring the existing routes in the canonical dictionary.

lmz committed:
Merged master into ywang
Resolved conflict in Build Standard Database.R

shimonisrael commented on make-uniform/production/Build Standard Database Functions.R:
```diff
@@ -63,6 +63,64 @@ get_nearest_station <- function(station_names_df, survey_records_df, operator_ke
 }
 
 
+# get_rail_names <- function(station_names_shp, 
```
@yuqiww, please remind me. This function was deprecated because rail is now handled elsewhere in the code?

yuqiww replied:
Yes. Please see the email titled "MTC Onboard Survey Scripts" from John Helsel on 2/19/2020.

yuqiww committed:
remove unnecessary variables from ACE and Solano County surveys

yuqiww committed:
Merge branch 'ywang' of https://github.com/BayAreaMetro/onboard-surveys into ywang

yuqiww committed:
make 'weekpart' standard variable coding for Solano County survey consistent with others

yuqiww committed:
remove unnecessary variable 'day_of_week'

lmz committed:
Remove .bak files

lmz committed:
Remove Readme.html

We use Readme.md

lmz commented:
Since there are so many diffs in this, I'm going to review the branch rather than the pull request and focus on the make-uniform\production process.

I'll also commit to the branch (and thus the pull request), as I already have started, a few clean-up items and update documentation as I read/understand things.

lmz committed:
Build Standard Database.Rmd was deprecated in favor of .R version

See commit 114ffe9d0d3dfcb167e026168df96f1646b942b4

yuqiww committed:
revise Marin Transit (2017) variable dictionary based on the new data from the consultant

yuqiww committed:
add Marin Transit (2017) transfer route canonical names

yuqiww committed:
Merge branch 'ywang' of https://github.com/BayAreaMetro/onboard-surveys into ywang

yuqiww committed:
add Marin Transit (2019) data into the code

yuqiww committed:
update local file directory for user 'ywang'

yuqiww committed:
update Napa (2019) preprocessing scipt based on the new data file

yuqiww committed:
make 'direction' variable consistent

yuqiww committed:
revise 'Access_mode'/'Egress_mode' for ACE (2019), Solano (2017), WestCAT (2017)

consolidate 'Access_mode' and 'Access_mode_other' into 'Access_mode_final', and 'Egress_mode', 'Egress_mode_other' into 'Egress_mode_final'

yuqiww committed:
revise 'Orig' and 'Dest' for ACE (2019)

consolidate 'Orig' and 'Orig_other' into 'Orig_final', and 'Dest', 'Dest_other' into 'Dest_final'

yuqiww committed:
conlidate 'Access_mode'/'Access_mode_other', 'Egress_mode'/'Egress_mode_other' in preprocessing

lmz commented:
I currently can't run the script because this file doesn't exist: 
**M:\Data\OnBoard\Data and Reports\Marin Transit\Final Data\marin transit_data file_final01222021_NO POUND OR SINGLE QUOTE.csv**

@yuqiww - I'm guessing you have this in your folder and it needs to be synced over?  Also, how come you're using a local copy of M:\Data\OnBoard\Data and Reports ?

yuqiww commented:
> I currently can't run the script because this file doesn't exist:
> **M:\Data\OnBoard\Data and Reports\Marin Transit\Final Data\marin transit_data file_final01222021_NO POUND OR SINGLE QUOTE.csv**
> 
> @yuqiww - I'm guessing you have this in your folder and it needs to be synced over? Also, how come you're using a local copy of M:\Data\OnBoard\Data and Reports ?

We're still sorting out some of the fields in the Marin Transit data. Would you please just comment out this operator for now?

yuqiww committed:
script to preprocess Marin Transit (2017) survey data

yuqiww commented:
> > I currently can't run the script because this file doesn't exist:
> > **M:\Data\OnBoard\Data and Reports\Marin Transit\Final Data\marin transit_data file_final01222021_NO POUND OR SINGLE QUOTE.csv**
> > @yuqiww - I'm guessing you have this in your folder and it needs to be synced over? Also, how come you're using a local copy of M:\Data\OnBoard\Data and Reports ?
> 
> We're still sorting out some of the fields in the Marin Transit data. Would you please just comment out this operator for now?

@lmz : the Marin data is now on M.

yuqiww committed:
code TNCs as a separate access_mode/egress_mode 'tnc'

yuqiww committed:
make canonical route name consistent for Petaluma transit and Santa Rosa Citybus routes

yuqiww committed:
separate 'tnc' from 'knr' for access_mode/egress_mode

create a new variable "path_access_recode" and "path_egress_recode" to represent the assumption that 'bike' = 'drive' and 'tnc' = 'knr' (which also = 'drive')

yuqiww committed:
fix bug in time period coding - midday (10am-3pm)

yuqiww committed:
add Petaluma (2018) data into the code

yuqiww committed:
label "don't know" as "Missing" for household income

yuqiww committed:
add Petaluma (2018) transfer routes to canonical_route_crosswalk

yuqiww committed:
add Petaluma (2018) variables to dictionary_for_standard_variable

yuqiww committed:
add Santa Rosa Citybus (2018) survey data to the code

yuqiww committed:
add Santa Rosa Citybus (2018) transfer routes to canonical_route_crosswalk

yuqiww committed:
add Santa Rosa Citybus (2018) survey variables to dictionary_for_standard_database

yuqiww committed:
add Santa Rosa Citybus (2018) survey data preprocess script

yuqiww committed:
Petaluma (2018) survey data preprocessing script

yuqiww committed:
fix typo in in-line comment

yuqiww committed:
treat "work-related" trip purpose as "other maintenance"

yuqiww committed:
correct default technology for SMART, ACE, Golden Gate Transit

yuqiww committed:
correct technology for Golden Gate Transit and SMART in canonical_route_crosswalk

yuqiww committed:
reverse the incorrect change on 'Missing___Missing Dummy Record"

yuqiww committed:
modify vehicles/workers/persons coding

consolidate categorical and noncategorical values; remove unnecessary code

yuqiww committed:
make 'persons' standard variable coding consistent in Dictionary

yuqiww committed:
make table() show 'na' counts

yuqiww committed:
create "technology_present" function

yuqiww committed:
use "technology_present" function to calculate technology present values

yuqiww committed:
fix bug in TriDelta (2019) transfer routes canonical name coding

yuqiww committed:
fix bug in TriDelta (2019) "number_transfers_" variable Dictionary

yuqiww committed:
move travel-model-specific variable configurations to a separate script

yuqiww committed:
capture all possible values in "vehicles"/"workers"/"persons"

yuqiww committed:
clean up 'time_string' and 'time_period' variable coding in Dictionary for Standard Database

yuqiww committed:
"daypart" defaults to 'time_period', then sets to 'time_string' if 'time_period' is NA

yuqiww committed:
add 'strata' for WETA (2019) and remove 'strata' for Muni (2017) in Dictionary

yuqiww committed:
bring in 'strata' field in WETA (2019) preprocessing

yuqiww committed:
revise added field names for Solano County (2017) to avoid using 'id'

shimonisrael committed:
Cleaned up tour code and added imputation purpose

shimonisrael committed:
More cleanup for tour purpose

shimonisrael committed:
Include test file and frequency statements

shimonisrael committed:
Update working directory location

yuqiww committed:
revise the technology of Golden Gate Transit bus as 'express bus'

shimonisrael committed:
Update TAP code to read in multiple modes per tap (TM2 TAPs)

shimonisrael committed:
Merge branch 'ywang' of https://github.com/BayAreaMetro/onboard-surveys into ywang

yuqiww committed:
update 'Napa Vine route 21' to 'express bus'

yuqiww committed:
Merge branch 'ywang' of https://github.com/BayAreaMetro/onboard-surveys into ywang

yuqiww committed:
revise WestCAT express bus technology

yuqiww committed:
add survey routes of multi-tech operators to canonical_route_crosswalk

yuqiww committed:
correct survey_route standard variable coding for Napa Vine (2014) survey

yuqiww committed:
add code to update survey_route technology for multi-technology operators

yuqiww committed:
script to pull survey_route raw data from multi-technology operators

Will be added to 'canonical_route_crosswalk', which will be used to update the route technology

yuqiww committed:
interim files used to add survey_routes of multi-tech operators to canonical_route_crosswalk

yuqiww committed:
create README.md for preprocessing scripts and interims files

yuqiww committed:
update preprocessing scripts names and README.md

shimonisrael committed:
Merge pull request #35 from BayAreaMetro/route-level-decomp

Route level decomp

yuqiww committed:
convert 'survey_routes' variable to noncategorical variables for all surveys

yuqiww committed:
remove unnecessary categorical_variable - canonical_route_crosswalk join

yuqiww committed:
use the noncategorical variable "survey_route_recode" for WestCAT

yuqiww committed:
truncate before/after transfers to no more than 3 for Muni and AC Transit

yuqiww committed:
use noncategorical survey_route variables for mulfi-technology survey_routes aggregation script

shimonisrael committed:
Create Muni Export for Georgia Tech.r

shimonisrael committed:
Merge branch 'ywang' of https://github.com/BayAreaMetro/onboard-surveys into ywang

shimonisrael committed:
Found some variables with unnecessary whitespace

yuqiww committed:
add 'weekpart' variable to Standard Dictionary

Add for the following surveys: County Connection (2019), LAVTA (2018), SF Muni (2017). TriDelta (2019), WestCAT (2017)

yuqiww committed:
create log file during code run

shimonisrael committed:
Updating documentation

shimonisrael committed:
Merge branch 'ywang' of https://github.com/BayAreaMetro/onboard-surveys into ywang

shimonisrael committed:
Fix day-part potion of the code and more documentation

shimonisrael committed:
Documenting marginal controls

yuqiww committed:
don't sync log files

yuqiww committed:
minor updates (see detailed description)

1) adjusted file directory, 2) consolidated fields, 3) deleted interim fields before exporting standardized data

yuqiww committed:
add detailed survey_variables

shimonisrael committed:
Update documentation for part where data is developed for PopSim inputs

shimonisrael committed:
Merge branch 'ywang' of https://github.com/BayAreaMetro/onboard-surveys into ywang

shimonisrael committed:
Document much shorter postprocess script

shimonisrael committed:
Initial commit.

shimonisrael committed:
Progress on weighting

shimonisrael committed:
Script is finalized

Script produces final weekday weights for Capitol Corridor survey.

yuqiww committed:
preprocess Capitol Corridor 2019 OD survey data

shimonisrael committed:
Fix documentation.

shimonisrael committed:
Merge branch 'ywang' of https://github.com/BayAreaMetro/onboard-surveys into ywang

yuqiww committed:
add Capitol Corridor 2019 OD survey canonical_route_crosswalk

yuqiww committed:
add Capitol Corridor 2019 OD survey standard dictionary

yuqiww committed:
add Capitol Corridor 2019 OD survey to the standard code

yuqiww committed:
Merge branch 'ywang' of https://github.com/BayAreaMetro/onboard-surveys into ywang

shimonisrael committed:
Update path for input file

shimonisrael committed:
Merge branch 'ywang' of https://github.com/BayAreaMetro/onboard-surveys into ywang

shimonisrael committed:
Include unspecified Fairfield locations in weighting scheme

shimonisrael committed:
Update documentation

shimonisrael committed:
Income summary for AC, LAVTA, Union City Transit
